hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/BatchConsumer.java

```diff
@@ -118,12 +128,37 @@ public void consume(Runnable signalsInterrupt) {
         }
     }
 
-    private void offerInflightOffsets(MessageBatch batch) {
-        batch.getPartitionOffsets().forEach(offsetQueue::offerInflightOffset);
+    private void commitIfReady() {
+        if (isReadyToCommit()) {
+            Set<SubscriptionPartitionOffset> offsetsToCommit = new HashSet<>();
+
+            for (Map.Entry<SubscriptionPartition, Long> entry : pendingOffsets.entrySet()) {
+                offsetsToCommit.add(new SubscriptionPartitionOffset(entry.getKey(), entry.getValue()));
+            }
+
+            if (!offsetsToCommit.isEmpty()) {
+                commit(offsetsToCommit);
+            }
+            lastCommitTime = Instant.now();
+        }
     }
 
-    private void offerCommittedOffsets(MessageBatch batch) {
-        batch.getPartitionOffsets().forEach(offsetQueue::offerCommittedOffset);
+    private boolean isReadyToCommit() {
+        return Duration.between(lastCommitTime, Instant.now()).toMillis() > commitPeriod.toMillis();
+    }
+
+    private void offerProcessedOffsets(MessageBatch batch) {
+        for (SubscriptionPartitionOffset offset : batch.getPartitionOffsets()) {
+            putOffset(offset);
+        }
+    }
+
+    private void putOffset(SubscriptionPartitionOffset offset) {
+        if (!pendingOffsets.containsKey(offset.getSubscriptionPartition())) {
```

Comment:
maybe this will be more readable:
```
        Long previousOffset = pendingOffsets.putIfAbsent(offset.getSubscriptionPartition(), offset.getOffset());
        if (previousOffset != null && previousOffset < offset.getOffset()) {
            pendingOffsets.put(offset.getSubscriptionPartition(), offset.getOffset());
        }
```

Reply:
or 
```java
pendingOffsets.compute(offset.getSubscriptionPartition(), (subscriptionPartition, maxOffset) ->
            maxOffset == null ? offset.getOffset() : Math.max(maxOffset, offset.getOffset())
        );
```